### PR TITLE
feat: Splits search api by resources to improve query time [cw-47]

### PR DESCRIPTION
**Pull request by nithindavid**
## Description

This PR moves away from a concise search API point url to three separate requests for messages, contacts and conversations.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by scmmishra — COMMENTED** [review]
@nithindavid the code overall seems good, will give it another look. In the UI I have some suggestions

#### When one of tab is toggled, we can hide the title?

Screenshot 2023-04-20 at 9 48 28 PM

Screenshot 2023-04-20 at 9 48 31 PM

#### When one of the sections is loading, we can show the title of the category, to indicate which part of the search category is pending

Screenshot 2023-04-20 at 9 48 54 PM

**Comment by nithindavid** [comment]
@scmmishra Thanks for the review! I made the changes.

**Review by iamsivin — COMMENTED** [review]
I tested it locally, and looks good to me. 
https://www.loom.com/share/706681df4e244fe399c512b1871ec60c

@nithindavid I found a small alignment issue.
image

File: app/javascript/dashboard/modules/search/components/SearchView.vue [hunk 1]
```diff
@@ -21,39 +21,44 @@
...
-              {{ $t('SEARCH.EMPTY_STATE_DEFAULT') }}
-            </p>
-          </div>
+        <div v-if="all.length">
+          <search-result-contacts-list
+            v-if="filterContacts"
+            :is-fetching="uiFlags.contact.isFetching"
+            :contacts="contacts"
+            :query="query"
+            :show-title="selectedTab === 'all'"
```
Comment (iamsivin) [inline app/javascript/dashboard/modules/search/components/SearchView.vue:30]: Can we move it to a computed property? We are using it in multiple places.
```suggestion
            :show-title="isSelectedTabAll"
```

**Review by iamsivin — COMMENTED** [review]
@nithindavid Please see the comment.

**Review by iamsivin — APPROVED** [review]
LGTM

**Review by iamsivin — APPROVED** [review]

**Review by scmmishra — APPROVED** [review]

**Pull request merged**